batch/batch/front_end/front_end.py

```diff
@@ -115,6 +115,49 @@ async def wrapped(request, userdata, *args, **kwargs):
     return wrapped
 
 
+async def _batch_billing_project_user(request, userdata):
+    app = request.app
+    db: Database = app['db']
+
+    batch_id = int(request.match_info['batch_id'])
+    user = userdata['username']
+
+    record = await db.select_and_fetchone(
+        '''
+SELECT id
+FROM batches
+LEFT JOIN billing_project_users ON batches.billing_project = billing_project_users.billing_project
+WHERE id = %s AND billing_project_users.`user` = %s;
+''',
+        (batch_id, user))
+
+    return record is not None
+
+
+def rest_billing_project_users_only(fun):
+    @rest_authenticated_users_only
+    @wraps(fun)
+    async def wrapped(request, userdata, *args, **kwargs):
+        permitted_user = await _batch_billing_project_user(request, userdata)
+        if not permitted_user:
+            raise web.HTTPUnauthorized()
```

Comment:
401 Unauthorized doesn't seem right. 401 [suggests that the client has not sent the proper authentication information](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401).

The question is then should we return 403 or 404. I think our policy is to always return 404.